ibis/client.py

```diff
@@ -280,13 +280,21 @@ def validate_backends(backends):
     return backends
 
 
-def execute(expr, limit='default', params=None, **kwargs):
-    (backend,) = validate_backends(list(find_backends(expr)))
+def resolve_backends(expr, client):
+    additional_backend = {client} if client is not None else set()
+    (backend,) = validate_backends(
+        set(find_backends(expr)).union(additional_backend)
+    )
+    return backend
+
+
+def execute(expr, limit='default', params=None, client=None, **kwargs):
+    backend = resolve_backends(expr, client)
     return backend.execute(expr, limit=limit, params=params, **kwargs)
 
 
-def compile(expr, limit=None, params=None, **kwargs):
-    (backend,) = validate_backends(list(find_backends(expr)))
+def compile(expr, limit=None, params=None, client=None, **kwargs):
+    backend = resolve_backends(expr, client)
```

Comment:
can you add a test that checks if client is specified and we *already* have the current backend set that it works correctly.

Reply:
are there cases where we should be raising an error here? do we need to check that client is a Client?